### ISSUE TITLE
[Inductor] functorch_dp_cifar10 accuracy regression in dynamic_inductor_torchbench

### ISSUE THREAD

**Issue by NikhilAPatel** [pytorch/pytorch#180740]
### Description

`functorch_dp_cifar10` fails accuracy checks in the `dynamic_inductor_torchbench` periodic CI job (CUDA, A10G):

```
functorch_dp_cifar10    FAIL:     accuracy=fail_accuracy, expected=pass
```

The error is a numerical accuracy regression in batch norm parameters:
```
Accuracy failed for key name layer2.0.bn2.bias
RMSE (res-fp64): 0.00606, (ref-fp64): 0.00145, shape=torch.Size([128])
```

### Details

- Only affects `dynamic_inductor_torchbench` (dynamic shapes + inductor), not static shapes
- Consistent across at least 4 consecutive periodic CI runs (April 17-18)
- Inference pass succeeds; only training accuracy fails
- The model compiles successfully (1 graph, 69 ops, 0 graph breaks)

### Expected CSV entry
`benchmarks/dynamo/ci_expected_accuracy/dynamic_inductor_torchbench_inference.csv`: `functorch_dp_cifar10,pass,0`

### CI Impact

Blocks `periodic-dynamo-benchmarks-test / test (dynamic_inductor_torchbench, 1, 2, linux.g5.4xlarge.nvidia.gpu)`.

cc @ezyang @penguinwu @bobrenjc93 @aditvenk @laithsakka @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

**Comment by benjaminglass1** [comment]
Documenting that I've been unable to repro this on any GPUs I have access to locally; I'm spinning up an Ampere instance on a cloud provider to check there (that's the closest thing I can get to the setup used in this CI job).

**Comment by NikhilAPatel** [comment]
Note: `functorch_dp_cifar10` has been baselined as `fail_accuracy` in `benchmarks/dynamo/ci_expected_accuracy/dynamic_inductor_torchbench_inference.csv` (PR #181299) to unblock CI.

When this issue is fixed, please update the baseline back to `pass,0`.

**Comment by benjaminglass1** [comment]
Documenting where I've gotten this to, given the weekend. The initial failure occurred on February 4, and appears to be associated with ea5b5f6174ad1a5e08950ddc8ab384fb3c68f63b. Specifically, copying over `_dynamo_weak_dynamic_indices` in `torch._dynamo.utils.clone_input` appears to expose a numerical issue that likely already existed with dynamic indexing. What that specifically is remains to be determined.

Additionally, there appears to be an underlying issue in the decomposition for `aten.native_group_norm.default` that shows up when running the same test with copying for `_dynamo_weak_dynamic_indices` patched out. Once I've traced them down a bit more I may split this into two issues to track each of these.

Also, this only seems to reproduce on Ampere or Ada GPUs (possibly newer, as well?). Older GPUs don't trigger either issue.

**Comment by benjaminglass1** [comment]
Final follow-up for this issue: the association with ea5b5f6 was a red herring; the real problem was in the decomposition for native_group_norm the whole time. #182907 reworks some of the numerical computation order in that decomposition, gets some extra speed, and fixes the accuracy bug.
